### ISSUE TITLE
Test failed: Assertion failed 'inVarToRegMap[varIndex] == REG_STK' in 'Xunit.ConsoleClient.Program:Main(ref):int' (IL size 79)

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#27539]
**Job:**
[coreclr-corefx-jitstress2-jitstressregs #20191029.1 Run Test Pri1 CoreFX Windows_NT x64 checked](https://dev.azure.com/dnceng/public/_build/results?buildId=405485)

**Detail:**
https://helix.dot.net/api/2019-06-17/jobs/3e84f706-3d36-4fe4-aa8e-4a6d411e010d/workitems/System.Threading.AccessControl.Tests/console

**OS & Arch:**
windows x64

**Mode:**
COMPlus_JitStress=2
COMPlus_JitStressRegs=0x80
COMPlus_TieredCompilation=0

**Log:**
```
Assert failure(PID 2172 [0x0000087c], Thread: 5640 [0x1608]): Assertion failed 'inVarToRegMap[varIndex] == REG_STK' in 'Xunit.ConsoleClient.Program:Main(ref):int' (IL size 79)

    File: F:\workspace\_work\1\s\src\jit\lsra.cpp Line: 4751
    Image: C:\dotnetbuild\work\3e84f706-3d36-4fe4-aa8e-4a6d411e010d\Payload\dotnet.exe

----- end Tue 10/29/2019  9:09:02.89 ----- exit code -1073740286
```

**Comment by AriNuer** [comment]
**Job:**
[coreclr-corefx-jitstressregs:20191104.1](https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_build/results?buildId=414529)

**Details:**
https://helix.dot.net/api/2019-06-17/jobs/f5eb4bc0-9b60-4ce8-b0ac-2a9836976503/workitems/Common.Tests/console

**OS & Arch:**
Linux x64

**Mode:**
export COMPlus_TieredCompilation=0
export COMPlus_DbgEnableMiniDump=1
export COMPlus_DbgMiniDumpName=$HELIX_DUMP_FOLDER/coredump.%d.dmp
export COMPlus_JitStressRegs=0x80

**Log:**
```
===========================================================================================================
~/work/f5eb4bc0-9b60-4ce8-b0ac-2a9836976503/Work/6701540c-03e9-4cf1-b6fd-b9cc6fd840c1/Exec ~/work/f5eb4bc0-9b60-4ce8-b0ac-2a9836976503/Work/6701540c-03e9-4cf1-b6fd-b9cc6fd840c1/Exec

Assert failure(PID 3025 [0x00000bd1], Thread: 3025 [0x0bd1]): Assertion failed 'inVarToRegMap[varIndex] == REG_STK' in 'Xunit.ConsoleClient.Program:Main(System.String[]):int' (IL size 79)

    File: /__w/14/s/src/jit/lsra.cpp Line: 4751
    Image: /home/helixbot/work/f5eb4bc0-9b60-4ce8-b0ac-2a9836976503/Payload/dotnet

Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.3025.dmp
Written 52756480 bytes (12880 pages) to core file
./RunTests.sh: line 161:  3025 Aborted                 "$RUNTIME_PATH/dotnet" exec --runtimeconfig Common.Tests.runtimeconfig.json xunit.console.dll Common.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing $RSP_FILE
~/work/f5eb4bc0-9b60-4ce8-b0ac-2a9836976503/Work/6701540c-03e9-4cf1-b6fd-b9cc6fd840c1/Exec
----- end Mon Nov 4 13:39:06 UTC 2019 ----- exit code 134 ----------------------------------------------------------
exit code 134 means SIGABRT Abort. Managed or native assert, or runtime check such as heap corruption, caused call to abort(). Core dumped.
```

**Comment by CarolEidt** [comment]
I believe that this is a duplicate of #27804 (actually vice-versa I guess since this was opened first). I have a fix in #27839 on which I'm running both coreclr-jitstressregs and coreclr-corefx-jitstressregs.

**Comment by CarolEidt** [comment]
Fixed with #27839
